src/PregWrapper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+class PregWrapper
+{
+    public static function replace($pattern, $replacement, $subject)
+    {
+        // TODO: check if "u" modifier is in the pattern and throw exception if so
+
+        $newSubject = preg_replace($pattern.'u', $replacement, $subject);
```

Comment:
I must admit that I was exactly thinking about this solution - try with `u`, and if it crashes try without `u`, yet didn't have time to see it working. I think this is the solution we shall aim to, if it works

:+1: :heart:

Reply:
I like the double-try too